cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -278,6 +279,7 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
                 .compose(state -> state.kafkaStatefulSet())
                 .compose(state -> state.kafkaRollingUpdate())
                 .compose(state -> state.kafkaScaleUp())
+                .compose(state -> state.kafkaPodsReady())
```

Comment:
not related to this PR, do you think that there is any hurt to have RollingUpdate and ScaleUp in the opposite order compared to zk? I don't think so but maybe just for code cleanness, we could set the same order.

Reply:
IIRC we did this intentionally. If the scale-up happens before the rolling update:
* It will first have to start with the old settings and then possibly be rolled-up immediately again. That might cost more time.
* Some of the information around external access are passed as env-vars. So it might not be able to actually start in the first place and we would need to be able to handle it.
So I do not think we can move it so easily. On the other hand this readiness check cost us minimal time and effort when the pods will be ready.